pkg/storage/utils/decomposedfs/node/locks.go

```diff
@@ -307,3 +312,37 @@ func (n *Node) hasLocks(ctx context.Context) bool {
 	_, err := os.Stat(n.LockFilePath()) // FIXME better error checking
 	return err == nil
 }
+
+func isLockModificationAllowed(ctx context.Context, oldLock *provider.Lock, newLock *provider.Lock) (bool, error) {
+	if oldLock.Type == provider.LockType_LOCK_TYPE_SHARED {
+		return true, nil
+	}
+
+	var forbidden bool = true
+	var err error
+
+	if oldLock.AppName != "" || newLock.AppName != "" {
+		if oldLock.AppName != newLock.AppName {
+			err = errtypes.PermissionDenied("cannot change holder (app name) when refreshing a lock")
```

Comment:
Does it make sense to continue after we encountered this error? Imho we could just `return false, err`

Reply:
done :+1: